[Feat] Button Component 제작

kyeoungwoon opened the pull request:
## #️⃣ Related Issues

> 연관된 모든 Issue를 작성해주세요.

- Issue #16
- Issue #107 

## 🧑‍💻 작업 내용

> 어떤 작업을 진행했는지 자세하게 작성해주세요.

- Figma 상의 Height 30px, 32px, 38px, 48px, 56px 에 대한 Button Components 제작 완료하였습니다.
- `styles/token/custom-style.css`를 생성하여 Figma 상의 `border: inline`에 대한 alias (`box-shadow`를 통해 구현) 를 제작하였습니다.

---

### 추가 작업 내용

PR #85 에서 작업한, 각 SVG 아이콘 중간에 추상화하여 컴포넌트화 한 작업물을 버튼 내에서 추가로 적용하였습니다.
Issue #107 에 자세하게 기술되어 있습니다.

---

## 💾 작업 결과 (선택)

> 사진, 영상 등을 첨부해주세요.

![image](https://github.com/user-attachments/assets/1347ed48-014b-4b83-9787-a633469afdc9)

SVG 추가 작업 (Issue #22) 이 완료되지 않아 SVG는 추가되어 있지 않습니다.

## 💬 리뷰 시 요청사항 (선택)

> Reviewer는 코드 리뷰의 코멘트에 코멘트를 강조하고 싶은 정도를 [Pn 규칙](https://blog.banksalad.com/tech/banksalad-code-review-culture/#%EC%BB%A4%EB%AE%A4%EB%8B%88%EC%BC%80%EC%9D%B4%EC%85%98-%EB%B9%84%EC%9A%A9%EC%9D%84-%EC%A4%84%EC%9D%B4%EA%B8%B0-%EC%9C%84%ED%95%9C-pn-%EB%A3%B0)에
> 맞춰서 표기해 주세요.

- 동작 및 activa 상태에 따른 확장성을 고려하였으나, 부족할 수 있으니 해당 부분을 중점적으로 판단 바랍니다.
- 추후 font system이 Figma 상에서 typography와 일치하게 되면 수정해야 합니다. (Issue 생성예정)

## 📝 Checklist

- [x] Reviewer를 추가했나요?
- [x] Convention을 준수했나요?

kyeoungwoon committed:
📦️ Package: Add clsx

kyeoungwoon committed:
💄 Ui: add primary-inactive css variable

color system에 추가 필요

kyeoungwoon committed:
✨ Feat: Create Button Components

Figma 상의 공용 버튼들은 모두 제작 완료하였습니다.

duwlsssss commented on .storybook/preview.ts:
```diff
@@ -1,3 +1,4 @@
+import '@/styles/globals.css';
```
P1: `@styles` 이런식으로 써야 import order 적용돼요!

S-Gihun replied:
P1: Cancle 과 Complete 부분은 DatePicker 때 사용돼서, 제가 임의로 onClick 설정해두었습니다. 이점 참고해서 다른 버튼들 onClick 이벤트만 추가해주시면 감사하겠습니다.

kyeoungwoon replied:
좋은 질문입니다!  
아래와 같이 `tsconfig.json`에 설정되어 있는 경우:

```json
"baseUrl": "./src",
"paths": {
  "@/*": ["*"],
  "@common/*": ["common/*"],
  "@features/*": ["features/*"],
  "@assets/*": ["assets/*"],
  "@styles/*": ["styles/*"]
}
```

---

## 각각 어떻게 동작하는가?

### 1. `import '@/styles/globals.css';`

- `@/*` → `*` 이므로  
  `@/styles/globals.css` → 실제로는 `src/styles/globals.css`를 의미합니다.

---

### 2. `import '@styles/globals.css';`

- `@styles/*` → `styles/*` 이므로  
  `@styles/globals.css` → 실제로도 `src/styles/globals.css`를 의미합니다.

---

## **결론**

**둘 다 똑같이 `src/styles/globals.css`를 가리킵니다.**  
즉, _동일한 파일을 import_ 하게 됩니다.

---

## **추가 설명**

- `@/styles/globals.css`:  
  `@/`는 `src/`의 최상위 경로를 의미합니다.
- `@styles/globals.css`:  
  `@styles/`는 `src/styles/`로 바로 매핑됩니다.

따라서 두 import 모두 결과적으로 `src/styles/globals.css`를 import합니다.

---

## **실무 팁**

- 보통 `@/`는 **최상위 기준**  
- `@styles/`, `@features/` 등은 **특정 폴더 바로 접근**  
- 팀 내 컨벤션에 따라 하나로 통일해서 쓰는 것이 유지보수에 좋습니다.

---

궁금한 점 더 있으면 언제든 질문해 주세요!

kyeoungwoon replied:
GPT 답변 추가해둡니다.
사실 스타일 적용이 되서 몰랐는데 컨벤션 상 혼돈 있을 수 있을 것 같아 수정 처리합니다!

duwlsssss reviewed (commented):
P2: 클릭 핸들러만 연결해주실 수 있을까요??

duwlsssss commented on src/common/components/buttons/30px/NextStepButton.tsx:
```diff
@@ -0,0 +1,23 @@
+import clsx from 'clsx';
+
+interface NextStepButtonProps {
+  isActive: boolean;
```
P3: 활성화 상태가 필요한 버튼들에는 `disabled props`로 
색 설정, button 태그에 disabled 속성을 연결하는 거 어떨까요??

kyeoungwoon replied:
disabled가 있다는걸 깜빡하고 있었네요. 변경해서 다시 Review 요청 올리겠습니다.

S-Gihun reviewed (commented):
P2: NextStepButton 만 스토리 작업이 이루어진 것으로 보여 button 스토리화를 전체적으로 진행할 것인지 여쭈어봅니다.

kyeoungwoon commented:
> P2: NextStepButton 만 스토리 작업이 이루어진 것으로 보여 button 스토리화를 전체적으로 진행할 것인지 여쭈어봅니다.

전체 스토리북화 할 예정이고, 따로 Issue 생성하여 진행 예정입니다.
해당 버튼 테스트 과정에서 임의로 추가해본 상태였습니다!

kyeoungwoon commented:
> P2: 클릭 핸들러만 연결해주실 수 있을까요??

@duwlsssss 

onClick에 대한 함수 말씀이시죠? 위 내용과 함께 적용하겠습니다

kyeoungwoon committed:
✨ Feat: add onClick props in all buttons

조이의 요청을 반영합니다.

kyeoungwoon committed:
♻️ Refactor: change prop name from isActive to disabled

조이의 요청에 따라 반영합니다.

kyeoungwoon committed:
🍻 Chore: merge develop

for conflict resolution

kyeoungwoon committed:
✨ Feat: add svg in icons

svg icon 중 download, plus는 stroke를 currentColor로 변경하였습니다.

kyeoungwoon committed:
🍻 Chore: delete unused type file and fix @common import

@/commomn -> @common

kyeoungwoon committed:
🍻 Chore: edit fill property in svg

필요없는 부분 삭제 및 패키지 재설치 진행하였음

kyeoungwoon committed:
🐛 Bug: fix build error

svg 로딩 시 활용하던 @svgr/webpack으로 인한 bundling error를 next.config.ts 파일을 수정하여 해결하였습니다.

kyeoungwoon commented:
수정한 내용 관련해서 sub issue 기록해두었습니다.
`으악 머리아파!`

kyeoungwoon committed:
🔀 Merge: merge develop

icon component 활용을 위해 merge 합니다.

kyeoungwoon committed:
✨ Feat: 2차 추가된 로고 추가 및 SVG 아이콘 추가

@buttons path alias가 추가되었습니다. Figma 상의 2차 추가된 아이콘 모두 추가되었습니다.

kyeoungwoon committed:
✨ Feat: 공용 버튼 컴포넌트들이 HTMLButtonElement의 속성을 상속받도록 수정

- HTML button을 사용하는 것과 코드 상에서 차이가 없도록 상속 설정을 하였습니다.
- 기존에 'type' 라는 필드명으로 구분하던 버튼 타입을 'buttonType'으로 변경하였습니다.
- 이는 HTMLButtonElement의 속성과 충돌을 방지하기 위함입니다.

리팩토링 필요한 부분
- 현재 각 컴포넌트에서 onClick 및 disabled를 직접 받지만, 이또한 props에 포함되어 있으므로
  구조분해할당할 필요 없이 그냥 사용해도 되지 않을까 하는 의견입니다.
- 사이즈가 동일한 버튼들은 앞으로 padding-x로 width를 유동적으로 조절할 수 있도록
  변경할 필요가 있습니다. -> 디자이너 분과 협업 필

kyeoungwoon committed:
🔨 Fix: SVG 컴포넌트화가 적용되지 않은 부분에 대한 수정

파일 구조가 미처 변경되지 않은 부분에 대한 수정.
VSCode source controller 이용하면 lefthook을 bypass 하는 사실 발견
여타 자잘한 수정 ...

kyeoungwoon committed:
🔨 Fix: 지금 건너뛰기 버튼에 크기 변경에 따른 text 변경이 적용되지 않은 부분 수정

그러합니다

S-Gihun commented on src/common/components/buttons/diverse-size/CancelButton/CancelButton.tsx:
```diff
@@ -0,0 +1,38 @@
+  return (
+    <button
+      className={clsx({
+        'border-stroke-200 inline-flexitems-center justify-center border bg-white hover:bg-gray-600': true,
```
P1: inline-flex 와 items-center 가 띄어쓰기 적용이 안 되어 있네요, 아마 이쪽 부분은 의도대로 동작되지 않을 것으로 예상됩니다. 수정 부탁드려요 : )

kyeoungwoon replied:
수정 완료했습니다!

S-Gihun reviewed (commented):
P5: Cancel 버튼 변동사항도 확인했습니다. 추후 Merge 후에 DatePicker 에 적용하도록 하겠습니다. `ButtonsCommonProps` 로 type 빼둔 것도 좋은 거 같습니다. P1 사항만 반영해주시면 감사하겠습니다. 고생 많으셨습니다.

kyeoungwoon committed:
🔨 Fix: tailwindcss class 내에 띄어쓰기 안 된 부분 수정

그러합니다

strfunctionk reviewed (approved):
P5: 고생하셨습니다
